tests/tests_firewall_fact.yml

```diff
@@ -0,0 +1,128 @@
+- name: Test firewall ansible fact
+  hosts: all
+  become: true
+  tasks:
+    - name: Test firewall ansible fact
+      block:
+        - name: Start with default configuration
+          include_role:
+            name: linux-system-roles.firewall
+          vars:
+            firewall:
+              - previous: replaced
+
+        # Test base fact (no custom settings applied)
+
+        - name: Get default ansible fact
+          include_role:
+            name: linux-system-roles.firewall
+          vars:
+            firewall:
+
+        - name: Fail if firewall_config is undefined
+          fail:
+            msg: Ansible fact firewall_config is not defined
+          when: firewall_config is undefined
+
+        - name: Fail if firewall_config custom exists
+          vars:
+            custom: custom
+          fail:
+            msg: custom should not exist if there is no custom configuration
+          when: custom in firewall_config.keys()
```

Comment:
`custom in firewall_config`

Also - you can use this form for `when` if the first thing in the `when` is a quoted string:
```
  when: "'custom' in firewall_config"
```
if you want to avoid the `custom` var